edx_proctoring/api.py

```diff
@@ -1424,7 +1424,7 @@ def _get_timed_exam_view(exam, context, exam_id, user_id, course_id):
     attempt_status = attempt['status'] if attempt else None
     has_due_date = True if exam['due_date'] is not None else False
     if not attempt_status:
-        if has_due_date_passed(exam['due_date']):
+        if has_due_date_passed(exam['due_date']) and not _is_self_paced_course(user_id, course_id):
```

Comment:
I suppose we could also update the exam due_date here as well if it has changed as you suggested this morning. What do you think?